### PR TITLE
Use tox for CI, get rid of travis's matrix and let tox handle stuff

### LINKED ISSUE #87: Upgrade Travis CI to support testing with varying dependency versions
A number of PRs going in that require testing with different versions of major dependency versions, like Tornado 4 - 5, Celery 3 - 4 (#85).

Using `tox` was suggested somewhere as an alternative to Travis matrix.

### PR DESCRIPTION
**Pull request by asherf**
Resolves #87

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,10 +33,10 @@ before_script:
   - sudo rm -f /etc/boto.cfg
 
 script:
-  - make test lint
+  - tox
 
 after_success:
-  if [ "$COVER" = "1" ]; then coveralls -v ; fi
+  coveralls -v
```
Comment (yurishkuro) [inline .travis.yml]: what would happen here, given that each step in the matrix would execute this? Is coveralls smart enough to merge coverage?

Comment (asherf) [reply .travis.yml]: yes it is.
But I think coverage may not be running now (when running pytest)..
still trying to figure out how to make sure it is enabled.
will push an update in the next few hours.

Comment (yurishkuro) [reply .travis.yml]: image

File: .travis.yml [hunk 2]
```diff
@@ -10,21 +10,10 @@ python:
...
-  include:
-  - python: '3.7'
-    env:
-      CELERY=4 COVER=1
+  - '3.7'
 
 install:
-  - make install-ci
-  - pip install celery~=${CELERY}.0
+  - pip install tox-travis
```
Comment (Jamim) [inline .travis.yml]:
```suggestion
  - pip install tox-travis coveralls
```

Comment (asherf) [reply .travis.yml]: oops... already pushed..

File: .travis.yml (hunk 1 above)
Comment (asherf) [reply .travis.yml]: 🤦‍♂ fixing

Comment (asherf) [reply .travis.yml]: https://coveralls.io/builds/24412824

**Pull request merged**